### PR TITLE
fix(resetSearchInput) change search box doesn't clear after selecting

### LINKED ISSUE #1484: Search box doesn't clear after selecting an option
In a tagging and multiple select input when you select an option the search input doesn't clear even when the reset-search-input is true.

The [documentation ](https://github.com/angular-ui/ui-select/wiki/ui-select) says: 
> reset-search-input : Clears the search box after selecting an option

More info here:
[Plunker](https://embed.plnkr.co/a8CLOUFTnPeYxBAF8JMB/)

### PR DESCRIPTION
**Pull request by Jefiozie**
fix(resetSearchInput) change search box doesn't clear after selecting

Previously when the closeOnSelect was set to false the resetSearchInput was not executed after selecting a item even when resetSearchInput was set to true. 

This fix will always execute the resetSearchInput when set to true even with the closeOnSelect set to false.

Fixes #1484

### CODE REVIEW

File: test/select.spec.js [hunk 1]
```diff
@@ -2758,9 +2758,267 @@ describe('ui-select tests', function() {
...
+
+        triggerKeydown(searchInput, Key.Enter);
+
+        expect($(el).scope().$select.search).toEqual('idontexist');
+      });
+    });
 
   });
 
+
```
Comment (user378230) [inline test/select.spec.js]: Can you remove this extraneous empty line?

**Comment by user378230** [comment]
Can you shorten the tests by using `createUiSelect()`? The indenting is a bit off too.

**Comment by Jefiozie** [comment]
I think this is better?

**Comment by user378230** [comment]
Much better, will review the code changes when I get a chance and merge, thanks! 😃

**Comment by Jefiozie** [comment]
Let me know if I need to make a change.

**Comment by Jefiozie** [comment]
Sorry @user378230  for all the commit stuff. New to git and didn't realize i was still pushing it here.

**Comment by user378230** [comment]
@Jefiozie think something went wrong here, can you rebase? or redo this PR? Sorry I've been away so haven't had a chance to look at PRs recently.

**Comment by Jefiozie** [comment]
I will rewrite it probably the best :smile:

**Comment by user378230** [comment]
I think also this will resolve #1384 ? :smiley:

**Comment by user378230** [comment]
And #1083? (even though there is no plunkr)

**Comment by Jefiozie** [comment]
@user378230  #1746 is the new PR for this issue. :smile:

**Pull request closed without merging**